sqlglot/optimizer/qualify_columns.py

```diff
@@ -73,21 +76,41 @@ def qualify_columns(
 
 def validate_qualify_columns(expression: E) -> E:
     """Raise an `OptimizeError` if any columns aren't qualified"""
-    unqualified_columns = []
+    all_unqualified_columns = []
     for scope in traverse_scope(expression):
         if isinstance(scope.expression, exp.Select):
-            unqualified_columns.extend(scope.unqualified_columns)
+            unqualified_columns = scope.unqualified_columns
+
             if scope.external_columns and not scope.is_correlated_subquery and not scope.pivots:
                 column = scope.external_columns[0]
-                raise OptimizeError(
-                    f"""Column '{column}' could not be resolved{f" for table: '{column.table}'" if column.table else ''}"""
-                )
+                for_table = f" for table: '{column.table}'" if column.table else ""
+                raise OptimizeError(f"Column '{column}' could not be resolved{for_table}")
+
+            if unqualified_columns and scope.pivots and scope.pivots[0].is_unpivot:
```

Comment:
what about like any(pivot.unpivot for pivot in scope.pivots)

Reply:
or a for loop on the pivots because you use it on 93

Reply:
Let's do the 1 to many transition progressively, don't think this will provide much value here because we still won't handle the many (un)pivot case ungracefully. At least this way I think it will work correctly for one and if there are >1 we'll raise due to ambiguous columns

Reply:
ok